### PR TITLE
Supported tokenize for Perl 5.24.1

### PR DESCRIPTION
**Pull request by Code-Hex**
This is a big change.
Because, I rewrote the code to add a new test each time.

大変期間が空いてしまい申し訳ありませんでした！  
テストの修正を行っていく度に、様々な問題を見つけたのでその度に修正していきました。  
とても大きな変更になりましたが、添削の方宜しくお願いします。  

## What I did ?

- Fixed to cause segfault
- Supported for lexing symbol variable like `*var`
- Fixed to scan heredoc
  - Supported lowercases heredoc's tag
  - Supported nested heredoc
  - Fixed to get line number
- Fixed to scan regex quotes like `q qq qx qw`
- Fixed to scan like `${^SOME_KEY}`
- Supported to can liken `${m} @{m} %{m} &{m} $#{m}`
- Fixed to scan `split //, $string`
- Added perl 5.24.1 tests
- Using new [Minilla](https://metacpan.org/pod/Minilla)

### CODE REVIEW

**Comment by Code-Hex** [comment]
Resolved issues:

- https://github.com/goccy/p5-Compiler-Lexer/issues/37
- https://github.com/goccy/p5-Compiler-Lexer/issues/59
- https://github.com/goccy/p5-Compiler-Lexer/issues/60
- https://github.com/goccy/p5-Compiler-Lexer/issues/62
- https://github.com/goccy/p5-Compiler-Lexer/issues/66
- https://github.com/goccy/p5-Compiler-Lexer/issues/67
- https://github.com/goccy/p5-Compiler-Lexer/issues/68
- https://github.com/goccy/p5-Compiler-Lexer/issues/69
- https://github.com/goccy/p5-Compiler-Lexer/issues/70
- https://github.com/goccy/p5-Compiler-Lexer/issues/76

Unchecked, but resolved

- https://github.com/goccy/p5-Compiler-Lexer/issues/79

Checked but maybe issue...

- https://github.com/goccy/p5-Compiler-Lexer/issues/72
- https://github.com/goccy/p5-Compiler-Lexer/issues/74

Not problem but matter for

- https://github.com/goccy/p5-Compiler-Lexer/issues/49

**Comment by Code-Hex** [comment]
@goccy ping

File: src/compiler/lexer/Compiler_lexer.cpp [hunk 1]
```diff
@@ -348,6 +350,14 @@ Token *Lexer::parseSyntax(Token *start_token, Tokens *tokens)
 		case LeftBracket: case LeftParenthesis:
 		case ArrayDereference: case HashDereference: case ScalarDereference:
 		case ArraySizeDereference: {
+			// Syntax error, It didn't close the brackets.
+			if (pos+1 == tokens->end()) {
+				fprintf(stderr,
```
Comment (goccy) [inline src/compiler/lexer/Compiler_lexer.cpp]: We needs reconsider error handling. if syntax error occurred, `tokenize()` 
should return error value and not force exit program.

Comment (Code-Hex) [reply src/compiler/lexer/Compiler_lexer.cpp]: I'm thinking it is like Go's error handling so, I agreed it.

**Pull request merged**